sql/src/main/java/io/crate/metadata/doc/DocTableInfoBuilder.java

```diff
@@ -106,8 +106,11 @@ private DocIndexMetaData docIndexMetaData() {
         }
         for (String concreteIndex : concreteIndices) {
             if (IndexMetaData.State.CLOSE.equals(metaData.indices().get(concreteIndex).getState())) {
-                throw new UnhandledServerException(
-                    String.format(Locale.ENGLISH, "Unable to access the partition %s, it is closed", concreteIndex));
+                if(!buildDocIndexMetaData(concreteIndex).closed()) {
```

Comment:
personal pref:  I'd move the buildDocIndexMetadata() call above the if statement, assign it to a var and use that in the if statement for better readability